libfreerdp/core/aad.c

```diff
@@ -607,30 +609,30 @@ static char* aad_create_jws_payload(rdpAad* aad, const char* ts_nonce)
 	return jws_payload;
 }
 
-static BOOL aad_update_digest(rdpAad* aad, EVP_MD_CTX* ctx, const char* what)
+static BOOL aad_update_digest(rdpAad* aad, WINPR_DIGEST_CTX* ctx, const char* what)
 {
 	WINPR_ASSERT(aad);
 	WINPR_ASSERT(ctx);
 	WINPR_ASSERT(what);
 
-	const int dsu1 = EVP_DigestSignUpdate(ctx, what, strlen(what));
-	if (dsu1 <= 0)
+	const BOOL dsu1 = winpr_DigestSign_Update(ctx, what, strlen(what));
+	if (!dsu1)
 	{
-		WLog_Print(aad->log, WLOG_ERROR, "EVP_DigestSignUpdate [%s] failed with %d", what, dsu1);
+		WLog_Print(aad->log, WLOG_ERROR, "winpr_DigestSign_Update [%s] failed", what);
 		return FALSE;
 	}
 	return TRUE;
 }
 
-static char* aad_final_digest(rdpAad* aad, EVP_MD_CTX* ctx)
+static char* aad_final_digest(rdpAad* aad, WINPR_DIGEST_CTX* ctx)
 {
 	char* jws_signature = NULL;
 
 	WINPR_ASSERT(aad);
 	WINPR_ASSERT(ctx);
 
 	size_t siglen = 0;
-	const int dsf = EVP_DigestSignFinal(ctx, NULL, &siglen);
+	const int dsf = winpr_DigestSign_Final(ctx, NULL, &siglen);
```

Comment:
Also change the call to `EVP_DigestSignFinal` below on line 650

Reply:
done